elastic-agent-client.proto

```diff
@@ -261,6 +261,8 @@ message UnitExpectedConfig {
     DataStream data_stream = 7;
     // Multiple streams per unit.
     repeated Stream streams = 8;
+    // Features configure, enable and disable optional features.
+    google.protobuf.Struct features = 9;
```

Comment:
Why `google.protobuf.Struct`? This is going to force every client to parse the feature as JSON effectively. Can we use a custom type with more structure?

I see some value in using struct because if we do not necessarily want to have to recompile the agent every time a feature gets added. It is maybe worth asking if that makes sense though, as features that only apply to particular integrations can just go in the integrations config. Can we think of features that would apply to multiple processes run by agent where the agent itself would require no changes?

If we wanted to allow the addition of features without recompiling the agent, we could still do something more structured like the following:

```protobuf
// Map instead of a repeated message so we can give each feature a name
map<string, Feature> features = 9;
```

```protobuf
message Feature {
  google.protobuf.Struct source = 1;
  // Pull out the enabled flag for convenience since it is always necessary.
  bool enabled = 1;
}

If we didn't want the ability to pass through features such that we expect to always have to change the agent, then something like this would work:

```protobuf
Features features = 9;
```

```protobuf
message Features {
  FQDNFeature fqdn = 1;
}

message FQDNFeature {
  bool enabled = 1;
}
```

Reply:
I question if feature flags should be per units.

I think it might be better to make this a top level key with a top level message type.

```
// A set of units and their expected states and configuration.
message CheckinExpected {
    // Units is the expected units the component should be running.
    repeated UnitExpected units = 1;
    // Agent info is provided only on first CheckinExpected response to the component.
    CheckinAgentInfo agent_info = 2;

    Features features = 3;
}
```